libjwt/jwt.c

```diff
@@ -1514,3 +1517,45 @@ unsigned int jwt_validate(jwt_t *jwt, jwt_valid_t *jwt_valid)
 
 	return jwt_valid->status;
 }
+
+typedef struct {
+	int error;
+	char *str;
+} jwt_exception_dict_t;
+
+jwt_exception_dict_t jwt_exceptions[] = {
+	/* { JWT_VALIDATION_SUCCESS, "SUCCESS" }, */
+	{ JWT_VALIDATION_ERROR, "General Failures" },
+	{ JWT_VALIDATION_ALG_MISMATCH, "Algorithm mismatch" },
+	{ JWT_VALIDATION_EXPIRED, "Token expired" },
+	{ JWT_VALIDATION_TOO_NEW, "Token future dated" },
+	{ JWT_VALIDATION_ISS_MISMATCH, "Issuer mismatch" },
+	{ JWT_VALIDATION_ISS_MISMATCH, "Subject mismatch" },
+	{ JWT_VALIDATION_AUD_MISMATCH, "Audience mismatch" },
+	{ JWT_VALIDATION_GRANT_MISSING, "Grant missing" },
+	{ JWT_VALIDATION_GRANT_MISMATCH, "Grant mismatch" },
+};
+
+char *jwt_dump_exceptions_string(int exceptions)
+{
+	int rc;
+	char *str = NULL;
+
+	if (exceptions == JWT_VALIDATION_SUCCESS)
+		return "SUCCESS";
+
+	for (int i = 0; i < ARRAY_SIZE(jwt_exceptions); i++)
+		if (jwt_exceptions[i].error & exceptions) {
+			if (str && (rc = __append_str(&str, ", ")))
+				break;
+
+			__append_str(&str, jwt_exceptions[i].str);
```

Comment:
Added it here and made sure that the "success" value was also done the same way. Switched to using 'goto fail' since other functions use gotos on failure too.